extensions/lifecycle/tasks/LifecycleObjectTask.js

```diff
@@ -0,0 +1,95 @@
+const async = require('async');
+const AWS = require('aws-sdk');
+
+const BackbeatTask = require('../../../lib/tasks/BackbeatTask');
+const AccountCredentials =
+    require('../../../lib/credentials/AccountCredentials');
+
+const attachReqUids =
+    require('../../replication/utils/attachReqUids');
+
+
+class LifecycleObjectTask extends BackbeatTask {
+    /**
+     * Process a lifecycle object entry
+     *
+     * @constructor
+     * @param {QueueProcessor} qp - queue processor instance
+     */
+    constructor(qp) {
+        const qpState = qp.getStateVars();
+        super({
+            retryTimeoutS: qpState.repConfig.queueProcessor.retryTimeoutS,
+        });
+        Object.assign(this, qpState);
+
+        this.s3Credentials = null;
+        this.s3Client = null;
+    }
+
+    _setupClients(log, done) {
+        this.s3Credentials = new AccountCredentials(this.authConfig, log);
+
+        const s3 = this.s3Config;
+        const transport = this.lcConfig.transport;
+        this.s3Client = new AWS.S3({
+            endpoint: `${transport}://${s3.host}:${s3.port}`,
+            credentials: this.s3Credentials,
+            sslEnabled: transport === 'https',
+            s3ForcePathStyle: true,
+            signatureVersion: 'v4',
+            httpOptions: { agent: this.httpAgent, timeout: 0 },
+            maxRetries: 0,
+        });
+
+        return done();
+    }
+
+    _executeAction(entry, log, done) {
+        const action = entry.action;
+        const { bucket, key, version } = entry.target;
+        const details = entry.details;
+        const reqParams = { Bucket: bucket, Key: key };
+        // default action is 'delete'
+        let reqMethod = 'deleteObject';
+
+        if (version) {
```

Comment:
If `version` is `undefined`, we can just set it in the `reqParams` object above and then we can remove this if block.